feature/system/gnmi/cliorigin/mixed_oc_cli_origin_support_test/README.md

```diff
@@ -0,0 +1,68 @@
+# gNMI-1.12: Mixed OpenConfig/CLI Origin
+
+## Summary
+
+Ensure that both CLI and OC configuration can be pushed to the device at the
+same time.
+
+## Procedure
+
+Note: this test is intended to cover only the case of pushing some configuration
+along with OC paths, since it is unknown what CLI configuration would be
+required in the emergency case that is covered by this requirement.
+
+*   TODO: Push base configuration to DUT specifying an interface configuration
+    for DUT port-1 and DUT port-2.
+
+*   Push non-overlapping mixed SetRequest specifying CLI for DUT port-1 and
+    OpenConfig for DUT port-2.
+
+    *   `origin: "cli"` containing vendor configuration.
+
+        ~~~
+        interface <DUT port-1>
+          description foo1
+        ~~~
+
+    *   `origin: ""` (openconfig, default origin) setting the DUT port-2
+        string value at `/interfaces/interface/config/description` to `"foo2"`.
+
+*   Validate the DUT port-1 and DUT port-2 descriptions through telemetry.
+
+*   Push overlapping mixed SetRequest specifying CLI before OpenConfig for
+    DUT port-1.
+
+    *   `origin: "cli"` containing vendor configuration.
+
+        ~~~
+        interface <DUT port-1>
+          description foo1
+        ~~~
+
+    *   `origin: ""` (openconfig, default origin) setting the DUT port-1
+        string value at `/interfaces/interface/config/description` to `"foo2"`.
+
+*   Validate that DUT port-1 description has `"foo2"`.
+
+*   TODO: Push overlapping mixed SetRequest specifying OpenConfig before CLI for
```

Comment:
You are introducing a behavior that is not agreed upon by the community with any overlap or order of operations here.  This is not what each implementation does so I suggest this be brought up in the wider OC community first before any codification - cc: @robshakir @dplore

Reply:
Ack

Reply:
@earies I will refer you to gNMI specification section 3.4:

https://github.com/openconfig/reference/blob/master/rpc/gnmi/gnmi-specification.md#34-modifying-state

> The order of the replace and update fields MUST be treated as significant within a single SetRequest message. If a single path is specified multiple times for a single operation (i.e., within update or replace), then the state of the target MUST reflect the application of all of the operations in order, even if they overwrite each other.

Reply:
Hey @earies this is to verify that what's the expected behavior in cases of such an overlap. The above specification does detail the requirements for such an operation. Do let us know if you have any more questions

Reply:
@liulk @hariramnat - Unfortunately there are still ambiguities that have not been thoroughly discussed and ratified when dealing with mixed schemas/origins.  The section you reference does reference the ordering of `Update` message application within a single `SetRequest` however there is an assumption that systems treat _all_ schemas equally with direct access to underlying objects when infact some implementations have had a preference model based off most explicit vendor native value wins.  OpenConfig (and any 3rd party modeling) is an abstraction layer built on top and can be done in a variety of ways.

For instance, you could send a `SetRequest` payload with the first `Update` containing `origin=openconfig` and the second `origin=cli` (or vice versa) but if there is a node that overlaps (lets pick on BGP ASN) then the most specific vendor native (cli in this case) always wins.

I still suggest further discussion as this is not how all implementations are built today.
Since this has always been per implementation behaviors from day 1 - this has significant architectural considerations

Reply:
@rgwilton , @rolandphung could you comment on this thread?  

I think it's most  important to understand the use case for why we want the behavior described in this PR.

The use case is to allow composition of an setRequest containing both OC and CLI origins.  We propose that the content be processed in the order it appears in the setRequest.  This allows flexible composition of configuration updates with OC and CLI.  While this example is contrived it is meant to generate questions about how a NOS will behave when given a setRequest with this content.  Our goal is to get deterministic configuration behavior from the NOS in the presence of multiple origins.

In [func TestOrderDependence](https://github.com/openconfig/featureprofiles/pull/474/files#diff-36bbeeef4c3dacadf0b003e85e03370579cd36775fe2d292d37d25bdd95638d4R72) there is an expectation that when pushing configuration using two different origins which overlap that the last to appear takes precedence.   

Note  @hariramnat , the way the test is written at the moment doesn't yet assert this, as it could also pass if the CLI always takes precedence.   Can you add another interface description update where origin CLI tries to set the value to some other unwanted value before the current two updates?

Reply:
@dplore, @liulk Done. Have added a testcase now to test whether the latest origin takes precedence or not.

Reply:
Thanks for the thoughtful comments!  Let's discuss in the openconfig community call.  The topic can be something like:  "Merge behavior between different gNMI origins in a gNMI setRequest"